File: Solutions/LeetCode/3536-MaximumProductOfTwoDigits/Solution.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int MaxProduct(int n) {
        int max1 = 0;
        int max2 = 0;
        while (n > 0) {
            int digit = n % 10;
            if (digit > max1) {
                max2 = max1;
                max1 = digit;
            } else if (digit > max2) {
                max2 = digit;
            }
            n /= 10;
        }
        return max1 * max2;
    }

    public static void Main(string[] args) {
        Solution sol = new Solution();
        var testCases = new (int n, int expected)[] {
            (31, 3),
            (22, 4),
            (124, 8)
        };

        for (int idx = 0; idx < testCases.Length; idx++) {
            var tc = testCases[idx];
            int result = sol.MaxProduct(tc.n);
            Console.WriteLine($"Test {idx + 1}: n={tc.n} -> Result={result} | Expected={tc.expected} | {(result == tc.expected ? "PASS" : "FAIL")}");
        }
    }
}

File: Solutions/LeetCode/3536-MaximumProductOfTwoDigits/Solution.py
class Solution:
    def maxProduct(self, n: int) -> int:
        digits = []
        while n > 0:
            digits.append(n % 10)
            n //= 10
        digits.sort(reverse=True)
        return digits[0] * digits[1]

if __name__ == '__main__':
    sol = Solution()
    test_cases = [
        (31, 3),
        (22, 4),
        (124, 8)
    ]
    for idx, (n, expected) in enumerate(test_cases, 1):
        result = sol.maxProduct(n)
        print(f"Test {idx}: n={n} -> Result={result} | Expected={expected} | {'PASS' if result == expected else 'FAIL'}")
